prow/plank/reconciler.go

```diff
@@ -235,7 +235,7 @@ func (r *reconciler) reconcile(pj *prowv1.ProwJob) (*reconcile.Result, error) {
 
 func (r *reconciler) terminateDupes(pj *prowv1.ProwJob) error {
 	pjs := &prowv1.ProwJobList{}
-	if err := r.pjClient.List(r.ctx, pjs, optNotCompletedProwJobs()); err != nil {
+	if err := r.pjClient.List(r.ctx, pjs, optPendingProwJobs()); err != nil {
```

Comment:
Isn't the "not completed" actually correct here? I understand this is called when new job is triggered e.g. by a PR push which needs to abort existing ones. In that case we would want to remove even the previous triggers which might be still waiting in the queue, no?

Reply:
Good point. I have changed it to use the `optPendingTriggeredJobsNamed` which will a) catch those as well b) only contains the jobs with the same name rather than all jobs